cpp/include/cugraph/prims/gather_edges.cuh

```diff
@@ -0,0 +1,710 @@
+void print_offset(raft::handle_t const& handle,
+                  GraphViewType graph_view,
+                  typename GraphViewType::vertex_type vertex_id,
+                  std::string str = "")
+{
+  using partition_t = cugraph::matrix_partition_device_view_t<typename GraphViewType::vertex_type,
+                                                              typename GraphViewType::edge_type,
+                                                              typename GraphViewType::weight_type,
+                                                              GraphViewType::is_multi_gpu>;
+  for (size_t i = 0; i < graph_view.get_number_of_local_adj_matrix_partitions(); ++i) {
+    auto matrix_partition = partition_t(graph_view.get_matrix_partition_view(i));
+    if ((vertex_id >= graph_view.get_local_adj_matrix_partition_row_first(i)) &&
+        (vertex_id < graph_view.get_local_adj_matrix_partition_row_last(i))) {
+      std::string offset_label =
+        str + std::string("\noffset ") +
+        std::to_string(graph_view.get_local_adj_matrix_partition_row_first(i));
+      debug::print(handle,
+                   matrix_partition.get_offsets() +
+                     (vertex_id - graph_view.get_local_adj_matrix_partition_row_first(i)),
+                   matrix_partition.get_offsets() +
+                     (vertex_id - graph_view.get_local_adj_matrix_partition_row_first(i)) + 2,
+                   offset_label,
+                   " ");
+    }
+  }
+}
+template <typename GraphViewType>
+void is_offset_sorted(raft::handle_t const& handle, GraphViewType graph_view, std::string str = "")
+{
+  using partition_t = cugraph::matrix_partition_device_view_t<typename GraphViewType::vertex_type,
+                                                              typename GraphViewType::edge_type,
+                                                              typename GraphViewType::weight_type,
+                                                              GraphViewType::is_multi_gpu>;
+  for (size_t i = 0; i < graph_view.get_number_of_local_adj_matrix_partitions(); ++i) {
+    auto matrix_partition = partition_t(graph_view.get_matrix_partition_view(i));
+    bool sorted =
+      thrust::is_sorted(handle.get_thrust_policy(),
+                        matrix_partition.get_offsets(),
+                        matrix_partition.get_offsets() + matrix_partition.get_major_size() + 1);
+    auto segment_offsets = graph_view.get_local_adj_matrix_partition_segment_offsets(i);
+    auto use_dcs         = segment_offsets
+                             ? ((*segment_offsets).size() >
+                        (cugraph::detail::num_sparse_segments_per_vertex_partition + 1))
+                             : false;
+
+    point(handle, "is_sorted ", static_cast<int>(sorted), i, use_dcs);
+  }
+}
+}  // namespace debug
+#endif
+namespace cugraph {
+
+namespace detail {
+
+/**
+ * @brief Calculate local out degrees of the sources belonging to the adjacency matrices
```

Comment:
And we're using major/minor in the detail namespace and source & destination (currently, we're mixing row&column as well, but in long run, I think we'd better not mix, and just use source & destination) in the public namespace, this same code can be used to compute in-degrees if the storage is transposed. This code computes major degrees (which are out-degrees in CSR + DCSR and in-degrees in CSC + DCSC).

Better rename this to `compute_local_major_degrees()` to be consistent with the functions in graph_impl.cuh (e.g. https://github.com/rapidsai/cugraph/blob/branch-22.04/cpp/src/structure/graph_view_impl.cuh#L86).